cluster/store/retry_schema.go

```diff
@@ -0,0 +1,152 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package store
+
+import (
+	"github.com/cenkalti/backoff/v4"
+	"github.com/weaviate/weaviate/cluster/utils"
+	"github.com/weaviate/weaviate/entities/models"
+	"github.com/weaviate/weaviate/usecases/sharding"
+)
+
+// retrySchema is used for retrying schema queries. It is a thin wrapper around
+// the original schema, separating retry logic from the actual operation.
+// Retry may be needed due to eventual consistency issues where
+// updates might take some time to arrive at the follower.
+type retrySchema struct {
+	schema *schema
+}
+
+func (rs retrySchema) ClassInfo(class string) (ci ClassInfo) {
+	rs.retry(func(s *schema) error {
+		ci = rs.schema.ClassInfo(class)
+		if !ci.Exists {
+			return errClassExists
+		}
+		return nil
+	})
+	return
+}
+
+// ClassEqual returns the name of an existing class with a similar name, and "" otherwise
+// strings.EqualFold is used to compare classes
+func (rs retrySchema) ClassEqual(name string) string {
+	return rs.schema.ClassEqual(name)
+}
+
+func (rs retrySchema) MultiTenancy(class string) models.MultiTenancyConfig {
+	return rs.metaClass(class).MultiTenancyConfig()
+}
+
+// Read performs a read operation `reader` on the specified class and sharding state
+func (rs retrySchema) Read(class string, reader func(*models.Class, *sharding.State) error) error {
+	return rs.retry(func(s *schema) error {
+		return rs.schema.Read(class, reader)
```

Comment:
**suggestion:** i think this shall be `return s.schema.Read(class, reader)` 🤔 or is it meant to come from the retryschema